src/radio-buttons.js

```diff
@@ -0,0 +1,37 @@
+import { hideHappyPath, showHappyPath } from "./happy-path";
+import { hideConformanceData, showConformanceData } from "./conformance";
+import { hideComplianceRules, showComplianceRules } from "./compliance-rules";
+
+let checkedKPIRadioButton = null;
+
+/**
+ * @param {BpmnVisualization} bpmnVisualization
+ */
+export function configureRadioButtons(bpmnVisualization) {
+    new KPIRadioButton("happy_path", () => showHappyPath(bpmnVisualization), () => hideHappyPath(bpmnVisualization));
+    new KPIRadioButton("conformance_data", () => showConformanceData(bpmnVisualization), () => hideConformanceData(bpmnVisualization));
+    new KPIRadioButton("compliance_rules", () => showComplianceRules(bpmnVisualization), () => hideComplianceRules(bpmnVisualization));
+
+    document.getElementById("reset_all").addEventListener("click", () => checkedKPIRadioButton?.hide());
+}
+
+class KPIRadioButton {
+    constructor(id, showCallback, hideCallback) {
+        this.hideCallback = hideCallback;
+
+        document.getElementById(id).addEventListener("click", function(){
```

Comment:
❓ did you try an arrow function instead? Generally it keep the right value for `this` which simplifies the syntax (no need to call `bind`).

Reply:
Yes, and it's not work because we want `this` to be equal to the instance of the RulaRadioButton class, and not to the not named function.